nvflare/edge/assessors/model_update.py

```diff
@@ -0,0 +1,336 @@
+        const_selection=False,
+    ):
+        Assessor.__init__(self)
+        self.persistor_id = persistor_id
+        self.persistor = None
+        self.current_model = None
+        self.current_model_version = 0
+        self.current_selection = {}
+        self.current_selection_version = 0
+        self.updates = {}  # model_version => _ModelState
+        self.available_devices = {}
+        self.used_devices = {}
+        self.num_updates_for_model = num_updates_for_model
+        self.max_model_version = max_model_version
+        self.max_model_history = max_model_history
+        self.device_selection_size = device_selection_size
+        self.min_hole_to_fill = min_hole_to_fill
+        self.device_reuse = device_reuse
+        self.const_selection = const_selection
+        self.update_lock = threading.Lock()
+        self.start_time = None
+        self.register_event_handler(EventType.START_RUN, self._handle_start_run)
+
+    def _handle_start_run(self, event_type: str, fl_ctx: FLContext):
+        engine = fl_ctx.get_engine()
+        self.persistor = engine.get_component(self.persistor_id)
+        if not isinstance(self.persistor, LearnablePersistor):
+            self.system_panic(reason="persistor must be a Persistor type object", fl_ctx=fl_ctx)
+            return
+        if self.persistor:
+            model = self.persistor.load(fl_ctx)
+
+            if not isinstance(model, ModelLearnable):
+                self.system_panic(
+                    reason=f"Expected model loaded by persistor to be `ModelLearnable` but received {type(model)}",
+                    fl_ctx=fl_ctx,
+                )
+                return
+
+            # Wrap learnable model into a DXO
+            self.current_model = model_learnable_to_dxo(model)
+            self.fire_event(AppEventType.INITIAL_MODEL_LOADED, fl_ctx)
+        else:
+            self.system_panic(reason="cannot find persistor component '{}'".format(self.persistor_id), fl_ctx=fl_ctx)
+            return
+
+    def start_task(self, fl_ctx: FLContext) -> Shareable:
+        self.start_time = time.time()
+        # empty base state to start with
+        base_state = BaseState(
+            model_version=0,
+            model=None,
+            device_selection_version=0,
+            device_selection={},
+        )
+        return base_state.to_shareable()
+
+    def _generate_new_model(self, fl_ctx: FLContext):
+        # New model generated based on the current global weights and all updates
+        new_model = {}
+        self.current_model_version += 1
+        old_model_versions = []
+
+        if self.current_model_version == 1:
+            # Initial global weights
+            new_model = self.current_model.data
+        else:
+            # Aggregate all updates
+            for v, ms in self.updates.items():
+                # FedBuff weight is 1/(1+staleness)^0.5
+                weight = 1 / (1 + (self.current_model_version - v) ** 0.5)
```

Comment:
so this ModelUpdateAssessor is actually implementing FedBuff?

Can you add paper reference in the `__init__` docstrings?